shoop/core/models/orders.py

```diff
@@ -214,9 +219,15 @@ class Order(models.Model):
     extra_data = JSONField(blank=True, null=True)
 
     # Money stuff
-    taxful_total_price = MoneyField(editable=False, verbose_name=_('grand total'), default=0)
-    taxless_total_price = MoneyField(editable=False, verbose_name=_('taxless total'), default=0)
-    display_currency = models.CharField(max_length=4, blank=True)
+    taxful_total_price = TaxfulPriceProperty('taxful_total_price_value', 'currency')
+    taxless_total_price = TaxlessPriceProperty('taxless_total_price_value', 'currency')
+
+    taxful_total_price_value = MoneyValueField(editable=False, verbose_name=_('grand total'), default=0)
+    taxless_total_price_value = MoneyValueField(editable=False, verbose_name=_('taxless total'), default=0)
+    currency = CurrencyField()
+    prices_include_tax = models.BooleanField()  # TODO: (TAX) Document Order.prices_include_tax
```

Comment:
Because order.taxful_total_price is a TaxfulPrice object, but order.taxful_total_price_value is a Decimal. Operating with Prices makes it more type safe and allows smarter rendering of this stuff in the templates (since Price has the taxfulness info and the currency).

Or what did you mean?